### PR TITLE
bugfix(#1625): somethings off in mainnet when we display ecorouting options

### PR DESCRIPTION
**Pull request by adamazad**
# Summary

So, this does not really fix anything because it turned out COW does not tap into Swapr's mainnet pools. And that's why COW always performs worse than other protocols. 

This PR, however, should improve COW's ranking because it uses execution price without fee to compare against other trades in the list.

### CODE REVIEW

**Comment by MilanVojnovic95** [comment]
Tested and looks ok to me 👍 
It is positioned much better among the results.

On PR Netlify preview:
image


On swapr.site:
image

**Comment by 0xVenky** [comment]
Hmm, the fee of 15% for COW still seems to be not right :(

File: src/lib/eco-router/api.ts [hunk 1]
```diff
@@ -71,164 +74,94 @@ export async function getExactIn(
...
+        }
+        if (platform.name === RoutablePlatform.ONE_INCH.name) {
+          return OneInchTrade.getQuote({
+            quoteCurrency: currencyOut,
+            amount: currencyAmountIn,
+            maximumSlippage,
+            recipient: receiver,
+            tradeType: TradeType.EXACT_INPUT,
+          })
+        }
```
Comment (bejzik8) [inline src/lib/eco-router/api.ts:155]: This is a switch statement material, definitely!

File: src/lib/eco-router/api.ts [hunk 2]
```diff
@@ -258,164 +191,91 @@ export async function getExactOut(
...
-
-    CoWTrade.bestTradeExactOut({
-      currencyAmountOut,
-      currencyIn,
-      maximumSlippage,
-      receiver,
-      user,
+        return undefined
+      }
     })
```
Comment (bejzik8) [inline src/lib/eco-router/api.ts:275]: This whole block is repeated above. I would advise creating helper function which would take care of both cases and which would ease the maintenance in the future considering we would have to update the logic on one place.

**Review by bejzik8 — APPROVED** [review]
Did a review, left some comments.

**Review by berteotti — APPROVED** [review]
Awesome improvements. Code looks good to me!

**Pull request merged**